fixed headlines in docs

OskarStark opened the pull request:
I am targeting this branch, because this is a doc fix.

Closes #159

greg0ire commented:
- "numeration" is not a word
- keep it under 50 chars please

OskarStark commented:
> "numeration" is not a word

fixed

> keep it under 50 chars please

fixed

greg0ire commented on Resources/doc/reference/orm.rst:
```diff
@@ -2,8 +2,8 @@
-Scenario A - Using Gedmo Doctrine Extensions
-============================================
+Option A - Using Gedmo Doctrine Extensions
+==========================================
```
@sstok suggest that this might be b/c we use `=` here, just like on line 3. Can you try reverting to Scenario, and use something else than `=` for subtitles?

OskarStark replied:
To be honest, i don't really understand what you mean?

greg0ire replied:
I mean changing all the `=` line 6 to something like `~` or `-`

OskarStark replied:
ok, I will try, but anyway, it makes no sense, that `Scenario` disappears

OskarStark replied:
Based on this: http://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#headings

```
Normally, there are no heading levels assigned to certain characters as the structure is determined from the succession of headings. However, it is better to stick to the same convention throughout a project. For instance:

# with overline, for parts
* with overline, for chapters
=, for sections
-, for subsections
^, for subsubsections
“, for paragraphs
```

`=` is a section, and therefore correct

greg0ire replied:
if `=` is a section then everything under line 3 is and it means we are dealing with a subsection => you should use `-`

greg0ire replied:
Or alternatively, change line 3 with `*`

OskarStark commented:
I changed the first headline and used `#` for underlining

Looks good to me:
![screenshot 2017-03-16 09 03 05](https://cloud.githubusercontent.com/assets/995707/23987005/79136288-0a27-11e7-9f31-6bcc7231e95a.png)

OskarStark commented:
Hmm, but all our other doc files starting with a headline like this:

```
First Headline
==========
```

let me propose something else

OskarStark commented:
I removed the `Scenario`-part from the headline and keep the correct headline hierachy.
I also added an introduction sentence with acnhor links down to the two available options

greg0ire commented on Resources/doc/reference/orm.rst:
```diff
@@ -2,8 +2,12 @@
+.. _gedmo_doctrine_extensions:
+
+Using Gedmo Doctrine Extensions
+===============================
```
Github properly uses an h2 here, although you are still using the same sign. I don't get how rst works, but maybe this is bug-based :confused:

OskarStark replied:
The symfony docs for example, all the time start like this for the first headline, normally we have to change all headlines. Anyway I would like to rework all the docs and bring them in common. But for now the numbering is fixed

wouterj replied:
reStructured text allows the following characters to be used as title underline: ``! " # $ % & ' ( ) * + , - . / : ; < = > ? @ [ \ ] ^ _ ` { | } ~``. From this big list, the recommended ones by reSt are: ``= - ` : . ' " ~ ^ _ * + #``

The order in which you use them isn't important. When a new underline char is found, the level will simply be increased by one (h2 -> h3). This hierarchy will then be saved and used throughout that *document*.

For consistency, the [Symfony doc standards](http://symfony.com/doc/current/contributing/documentation/standards) defines this rule: *"The following characters are chosen for different heading levels: level 1 is = (equal sign), level 2 - (dash), level 3 ~ (tilde), level 4 . (dot) and level 5 " (double quote);"*

greg0ire replied:
> The order in which you use them isn't important. When a new underline char is found, the level will simply be increased by one (h2 -> h3). This hierarchy will then be saved and used throughout that document.

@wouterj I knew  that part, what I don't understand is why an `h2` is used although the same char is used. 

> normally we have to change all headlines. Anyway I would like to rework all the docs and bring them in common

I think we should rework the docs too, in one big PR. Thank you for proposing @OskarStark 

> For consistency, the Symfony doc standards defines this rule: "The following characters are chosen for different heading levels: level 1 is = (equal sign), level 2 - (dash), level 3 ~ (tilde), level 4 . (dot) and level 5 " (double quote);"

We should do the same so that it looks familiar to sf docs contributors IMO.

wouterj replied:
> @wouterj I knew that part, what I don't understand is why an h2 is used although the same char is used.

I think that's because *"As a general rule, a document can only have one top level headline."*, however I couldn't find it this quickly in the reSt specs.

greg0ire replied:
You're probably right. If it isn't in the specs, we should probably stay away from it though.

OskarStark replied:
Thanks for the detailed feedback @wouterj 👍 

> I think we should rework the docs too, in one big PR. Thank you for proposing @OskarStark

yes, will do this soon!

> For consistency, the Symfony doc standards defines this rule: "The following characters are chosen for different heading levels: level 1 is = (equal sign), level 2 - (dash), level 3 ~ (tilde), level 4 . (dot) and level 5 " (double quote);"
> 
> As a general rule, a document can only have one top level headline.

@greg0ire an you put this general rule (which should be applied to all new docs from now on) to the `CONTRIBUTING.md`?
And *yes* we should follow these symfony-docs rules too

greg0ire replied:
> @greg0ire an you put this general rule (which should be applied to all new docs from now on) to the CONTRIBUTING.md?

Adding that to my todo list

dmarkowicz replied:
I added overline = to first headline in https://github.com/sonata-project/SonataTranslationBundle/pull/157/files#diff-046a0b7c44e8854e46ceffe912928724R1 to make it H1 without rewriting all other headlines below. According to http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#sections. This probably caused some confusion 😳
```
== <----- 
H1
==

H2
==
```
#159 refers to an outdated version of the documentation from master branch which is not merged with 3.x.

greg0ire replied:
Oh right I didn't even notice it

greg0ire replied:
I added that contribution to [a PR I already have](https://github.com/sonata-project/dev-kit/pull/203/files#diff-55aa24e78df8010aea154e9506cab16dR97)

greg0ire commented:
Ok let's go with that then! Thanks!
